### PR TITLE
Refactor/union arguments

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

Union input types refactored, moving away from `${fieldName}_${concreteType}` to a nested object.

Also fixed a variety of bugs/introduced missing functionality, included but not limited to:

- Nested union connect/disconnect
- Fixed where input for nested update for union types

# Issue

#192 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: docs/asciidoc/type-definitions/unions-and-interfaces.adoc [hunk 1]
```diff
@@ -5,93 +5,118 @@ Unions and interfaces are abstract GraphQL types that enable a schema field to r
...
-  }
 }
 
 ----
 
 
 == Interface Types
 
-Using interface types will give you no real database support therefore no; query, update, delete, filter support. But instead used as a language feature to safeguard your schema. Great for when dealing with repetitive or large schemas you can essentially put "The side railings up".
+Using interface types will give you no database support, therefore there is no support for querying, updating, deleting, filtering. But instead used as a language feature to safeguard your schema. Great for when dealing with repetitive or large schemas you can essentially put "The side railings up".
```
Comment (danstarns) [inline docs/asciidoc/type-definitions/unions-and-interfaces.adoc:122]: I think the messaging needs to change here now.

**Review by danstarns — CHANGES_REQUESTED** [review]
Could we add a test case if upserting two unions at once ? Just to cover the many `refNodes`

**Comment by darrellwarde** [comment]
> Could we add a test case of upserting two unions at once? Just to cover the many `refNodes`

All done!

**Review by danstarns — APPROVED** [review]
Nice work on this, thank you for getting me out of the weeds with my over refactored attempt at this. 🍻

**Pull request merged**